resources/files/windows/PuppetBolt/PuppetBolt.psm1

```diff
@@ -8,8 +8,22 @@ $env:RUBY_DIR = $env:BOLT_BASEDIR
 $env:SSL_CERT_FILE = "$($env:BOLT_BASEDIR)\ssl\cert.pem"
 $env:SSL_CERT_DIR = "$($env:BOLT_BASEDIR)\ssl\certs"
 
+function ConvertTo-BoltArgs {
+    param(
+        [psobject]$inputObject
+    )
+
+    $outputArgs = @()
+
+    foreach($arg in $inputObject){
+        $outputArgs += [string]$arg.replace('"','"""')
+    }
+
+    Write-Output $outputArgs
+}
+
 function bolt {
-    &$env:RUBY_DIR\bin\ruby -S -- $env:RUBY_DIR\bin\bolt $args
+    &$env:RUBY_DIR\bin\ruby -S -- $env:RUBY_DIR\bin\bolt (ConvertTo-BoltArgs -inputObject $args)
```

Comment:
I don't think the type casting is necessary, because `-replace` is already smart enough to do the right thing.

```
❯ 1, 'foo', '"bar"', @(1, 2, 3), (@{'foo' = 'bar'; 'bar' = 1} | ConvertTo-Json) -replace '"', '"""'
1
foo
"""bar"""
1 2 3
{
  """bar""": 1,
  """foo""": """bar"""
}
```

Reply:
Need to run some more tests, but we should be able to do away with `ConvertTo-BoltArgs` and write this simply as:


```ruby
&$env:RUBY_DIR\bin\ruby -S -- $env:RUBY_DIR\bin\bolt ($args -replace '"', '"""')
```

Reply:
Would be hilarious if it turned out to be even simpler than I thought it would be. I think this has a good chance of working though since the only thing that can come in through ```$args``` is going to be a string anyway.

Reply:
I modified this as suggested and tested it in the scenarios I detailed earlier, and it does indeed still work as expected.